### PR TITLE
Binary serialization

### PR DESCRIPTION
**Pull request by GERey**

### CODE REVIEW

File: stack/corepersistence/collection/src/main/java/org/apache/usergrid/persistence/collection/serialization/impl/MvccEntitySerializationStrategyImpl.java [hunk 1]
```diff
@@ -318,26 +327,33 @@ public ByteBuffer toByteBuffer( final EntityWrapper wrapper ) {
                 builder.addBytes( STATE_PARTIAL );
             }
 
-            builder.addBytes( SER.toByteBuffer( wrapper.entity.get() ));
+            try {
+                builder.addBytes( mapper.writeValueAsBytes( wrapper.entity.get() ) );
+            }
+            catch ( JsonMappingException e ) {
```
Comment (tnine) [inline stack/corepersistence/collection/src/main/java/org/apache/usergrid/persistence/collection/serialization/impl/MvccEntitySerializationStrategyImpl.java]: Don't swallow exceptions.  If we get something weird here, wrap it in a CollectionRuntimeException and re-throw it.  We always want to signal to the caller something failed if we had a problem.

File: stack/corepersistence/collection/src/main/java/org/apache/usergrid/persistence/collection/serialization/impl/MvccEntitySerializationStrategyImpl.java [hunk 2]
```diff
@@ -346,7 +362,20 @@ public EntityWrapper fromByteBuffer( final ByteBuffer byteBuffer ) {
...
+
+            try {
+
+                byte[] array = jsonBytes.array();
+                int start = jsonBytes.arrayOffset();
+                int length = jsonBytes.remaining();
+                storedEntity = mapper.readValue( array,start,length,Entity.class);
+            }
+            catch ( IOException e ) {
+                e.printStackTrace();
```
Comment (tnine) [inline stack/corepersistence/collection/src/main/java/org/apache/usergrid/persistence/collection/serialization/impl/MvccEntitySerializationStrategyImpl.java]: Same thing

File: stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/utils/JsonUtils.java [hunk 3]
```diff
@@ -57,7 +59,7 @@
 
 
     static {
-        indentObjectMapper.getSerializationConfig().set( Feature.INDENT_OUTPUT, true );
+        indentObjectMapper.getSerializationConfig().with( SerializationFeature.INDENT_OUTPUT );//set( SerializationConfig.Feature.INDENT_OUTPUT, true );
```
Comment (tnine) [inline stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/utils/JsonUtils.java]: Can we shut this off?  I know it increases runtime, and we don't care if they're pretty when they're stored.

Comment (GERey) [inline stack/corepersistence/queryindex/src/main/java/org/apache/usergrid/utils/JsonUtils.java]: I'll look into shutting it off. I was just updating it for use with the new jackson.

**Pull request merged**